lib/rubocop/cop/lint/useless_constant_scoping.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for useless constant scoping. Private constants must be defined using
+      # `private_constant` or `class << self`. Even if `private` access modifier is used,
+      # it is public scope despite its appearance.
+      #
+      # It does not support autocorrection due to behavior change and multiple ways to fix it.
+      # Or a public constant may be intended.
+      #
+      # @example
+      #
+      #   # bad
+      #   class Foo
+      #     private
+      #     PRIVATE_CONST = 42
+      #   end
+      #
+      #   # good
+      #   class Foo
+      #     PRIVATE_CONST = 42
+      #     private_constant :PRIVATE_CONST
+      #   end
+      #
+      #   # good
+      #   class Foo
+      #     class << self
+      #       private
+      #       PRIVATE_CONST = 42
+      #     end
+      #   end
+      #
+      #   # good
+      #   class Foo
+      #     PUBLIC_CONST = 42 # If private scope is not intended.
+      #   end
+      #
+      class UselessConstantScoping < Base
+        MSG = 'Useless constant scope to `private` access modifier.'
```

Comment:
Isn't the access modifier useless in the constant scope instead of the other way around? I guess one can argue both are true, but to me it seems that "Useless `private` access modifier for constant scope." reads a bit better.